From ab5e680f595d2abc64a63205f544789a25d42e50 Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Sat, 20 Nov 2021 16:47:27 +0200
Subject: [PATCH] [SW-0] drivers: hl_npu: reset psoc as part of reset

Currently, ARCs are reset as part of hard reset and WD reset so that
bootrom flow re-starts in secureboot flow mode. However, AXI_ERR is
observed in PSOC when hard reset is issued without PSOC being reset.
It is recommended by HW team to reset PSOC as part of reset when ARCs
are reset. This patch updates mask for hard-reset and WD reset to
reset PSOC.

Change-Id: I1339e224d01d8c42e2f11d9676c171eb9eea887a
---

diff --git a/arch/gaudi2/common/reset.c b/arch/gaudi2/common/reset.c
index 844f784..3de8820 100644
--- a/arch/gaudi2/common/reset.c
+++ b/arch/gaudi2/common/reset.c
@@ -21,7 +21,7 @@
 };
 
 static const struct gaudi2_rst_mask sw_rst[] = {
-	{0x0, mmPSOC_RESET_CONF_PSOC_SW_RST_CFG},
+	{0x1, mmPSOC_RESET_CONF_PSOC_SW_RST_CFG},
 	{0x1, mmPSOC_RESET_CONF_CPU_SW_RST_CFG},
 	{0x3, mmPSOC_RESET_CONF_ARC_SW_RST_CFG},
 	{0xF, mmPSOC_RESET_CONF_SIF_SW_RST_CFG},
